src/LP/LP.jl

```diff
@@ -51,7 +103,12 @@ function write_function(io::IO, model::Model, func::MOI.ScalarAffineFunction{Flo
                 print(io, term.coefficient < 0 ? " - " : " + ")
                 Base.Grisu.print_shortest(io, abs(term.coefficient))
             end
-            print(io, " ", MOI.get(model, MOI.VariableName(), term.variable_index))
+
+            varname = MOI.get(model, MOI.VariableName(), term.variable_index)
+            if !verifyname(varname)
+                varname = correctname(varname)
+            end
```

Comment:
We should probably get all the variables at the start, correct all the names, and store the variable -> corrected name in a dict to avoid recomputing the corrected name every time.

Reply:
I just pushed a commit with this implemented. 

I'm thinking about doing the same with warnings, to avoid showing the same warning over and over again (I have a model with many variables named `sth[Edge 1 => 2]`, so I have the exact same message a zillion times when exporting the LP file). So, there would be a warning for [, another one for ^, etc. What do you think of it?

Reply:
Isn't that is what the `maxlog` parameter is for? 

Another option is to add some options, like I've done for MOF:
https://github.com/odow/MathOptFormat.jl/blob/master/src/MOF/MOF.jl

Reply:
Not exactly, as I understand it: it only checks if the exact same message has already been printed (but the source code is a tad too complicated for me to be sure about it).

Reply:
Adding options is probably the best way to go then.